tests/0075-retry.c

```diff
@@ -177,6 +177,7 @@ static void do_test_low_socket_timeout(const char *topic) {
         test_conf_set(conf, "socket.timeout.ms", "1000");
         test_conf_set(conf, "socket.max.fails", "12345");
         test_conf_set(conf, "retry.backoff.ms", "5000");
+        test_conf_set(conf, "retry.backoff.max.ms", "5000");
```

Comment:
Since we're explicitly increasing the value of retry.backoff.ms here, the new property retry.backoff.max.ms will cause retry.backoff.ms to reset to 1000 and log a warning, identical to java client behaviour.

we can either add this extra conf or change the times inside set_delay(), which assume the backoff will be a constant 5000ms

we can mention this in upgrade considerations that if you're depending on retry behaviour times in your code, then it might be a problem, but it's not something a user should be doing. But for implementing this KIP, retry times will be changing, it's necessary

Reply:
The test change is ok, it was a general thinking about the KIP but I guess we can just mention it in upgrade considerations and it won't cause problems (maybe it can solve some misconfiguration problems instead)